docs/remote_desktop_explainer.md

```diff
@@ -0,0 +1,53 @@
+
+Remote desktop web applications let users interact with the (native) desktop environment of a remote host from a local client (web) application.
+
+A user may want to use a WebAuthn authenticator connected to their local client in order to sign into a website in a browser running on a remote host. But users typically cannot physically access the remote desktop host, so it's impossible for them to plug in a USB security key or to pass the biometric challenge of a built-in platform authenticator. Instead, we propose a model for web-based remote desktop clients to execute remote WebAuthn requests locally.
+
+## Proposal
+
+We assume the remote desktop software is somehow able to intercept the WebAuthn request occurring on the remote host browser and forward that request to the local client web app. We would like the remote desktop client to be able to execute the forwarded WebAuthn request locally on behalf of the original Relying Party's origin. 
+
+To achieve this, the remote desktop client would invoke the WebAuthn API with the same PublicKeyRequestOptions dictionary as the original host, but add an additional extension that indicates it wants to act on behalf of the remote caller origin.
+
+```
+// The remote desktop client (https://myrdc.example) calls WebAuthn with
+// the PublicKeyRequestOptions of the original request issued by
+// https://accounts.example.com and forwarded from the remote desktop host.
+navigator.credentials.get({publicKey: {
+    challenge: …,
+    // Note that https://myrdc.example can't regularly claim this RP ID.
+    rpId: 'example.com',
+    allowCredentials: { … },
+    extensions: {
+        …,
+        // Additional extension inserted by https://myrdc.example to act on 
+        // behalf of https://accounts.example.com.
+        remoteDesktopClientOverride: {
+            origin: 'https://accounts.example.com',
+            sameOriginWithAncestor: false,
+        },
+    },
+}});
+```
+
+The two extension members, `origin` and `sameOriginWithAncestors`, replace the arguments of the same names when invoking `PublicKeyCredential`'s `[[[DiscoverFromExternalSource]]` internal method](https://w3c.github.io/webauthn/#dom-publickeycredential-discoverfromexternalsource-slot). This causes the user agent to match the scope of the RP ID against the overridden origin (https://accounts.example.com), rather than the origin of the Relying Party (https://myrdc.example). The supplied values are also used to assemble the `[CollectedClientData` dictionary](https://w3c.github.io/webauthn/#dictionary-client-data) that the authenticator signs over when generating an assertion. (The remaining `CollectedClientData` members can be inferred from context and therefore don't need to be injected.) To the Relying Party, the response should be indistinguishable from one that would have been generated without forwarding.
```

Comment:
A few square brackets are misaligned...
```suggestion
The two extension members, `origin` and `sameOriginWithAncestors`, replace the arguments of the same names when invoking [`PublicKeyCredential`'s `[[[DiscoverFromExternalSource]]` internal method](https://w3c.github.io/webauthn/#dom-publickeycredential-discoverfromexternalsource-slot). This causes the user agent to match the scope of the RP ID against the overridden origin (https://accounts.example.com), rather than the origin of the Relying Party (https://myrdc.example). The supplied values are also used to assemble the [`CollectedClientData` dictionary](https://w3c.github.io/webauthn/#dictionary-client-data) that the authenticator signs over when generating an assertion. (The remaining `CollectedClientData` members can be inferred from context and therefore don't need to be injected.) To the Relying Party, the response should be indistinguishable from one that would have been generated without forwarding.
```